File: client/src/features/orders/interfaces/order.ts
import { OrderItem, ShippingAddress } from ".";

interface Order {
    id: number;
    buyerId: string;
    shippingAddress: ShippingAddress;
    orderDate: string;
    orderItems: OrderItem[];
    subtotal: number;
    deliveryFee: number;
    orderStatus: string;
    total: number;
}

export default Order;

File: api/Entities/OrderAggregate/Order.cs
using System;
using System.Collections.Generic;
using api.Entities.OrderAggregate.Constants;

namespace api.Entities.OrderAggregate
{
    public class Order
    {
        public int Id { get; set; }
        public string BuyerId { get; set; }
        public ShippingAddress ShippingAddress { get; set; }
        public DateTime OrderDate { get; set; } = DateTime.Now;
        public List<OrderItem> OrderItems { get; set; }
        public long Subtotal { get; set; }
        public long DeliveryFee { get; set; }
        public OrderStatus OrderStatus { get; set; } = OrderStatus.Pending;
        public string PaymentIntentId { get; set; }
        public long GetTotal()
        {
            return Subtotal + DeliveryFee;
        }
    }
}